api/rest/client/methods.go

```diff
@@ -172,3 +178,106 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+type StatusFilter struct {
+	In, Out chan api.GlobalPinInfo
+	Done    chan struct{}
+	Err     chan error
+}
+
+func NewStatusFilter() *StatusFilter {
+	return &StatusFilter{
+		In:   make(chan api.GlobalPinInfo),
+		Out:  make(chan api.GlobalPinInfo),
+		Done: make(chan struct{}),
+		Err:  make(chan error),
+	}
+}
+
+type FilterParams struct {
+	*cid.Cid
+	Local     bool
+	Target    api.TrackerStatus
+	CheckFreq time.Duration
+}
+
+func (sf *StatusFilter) Filter(ctx context.Context, fp FilterParams) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				sf.Err <- ctx.Err()
+				return
+			case gblPinInfo := <-sf.In:
```

Comment:
since you are closing `sf.In`, better check `gblPinInfo, more := <- sf.In` and exit if more == false.